src/App.js

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react'
+import { Routes, Route, createSearchParams, useSearchParams, useNavigate } from "react-router-dom"
+import { useDispatch, useSelector } from 'react-redux'
+import 'reactjs-popup/dist/index.css'
+import { fetchMovies } from './data/moviesSlice'
+import { ENDPOINT_SEARCH, ENDPOINT_DISCOVER, ENDPOINT, API_KEY } from './constants'
+import Header from './components/Header'
+import Movies from './components/Movies'
+import Starred from './components/Starred'
+import WatchLater from './components/WatchLater'
+import YouTubePlayer from './components/YoutubePlayer'
+import './app.scss'
+
+const App = () => {
+
+  const state = useSelector((state) => state)
+  const { movies } = state  
+  const dispatch = useDispatch()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const searchQuery = searchParams.get('search')
+  const [videoKey, setVideoKey] = useState()
+  const [isOpen, setOpen] = useState(false)
+  const navigate = useNavigate()
+  
+  const closeModal = () => setOpen(false)
+  
+  const closeCard = () => {
```

Comment:
It seems this function isn't currently utilized in the codebase. To ensure code cleanliness and maintainability, would you prefer removing this unused function, or is there an intended functionality you plan to implement in the future?